test/core/utils/dq-element.js

```diff
@@ -4,227 +4,258 @@ describe('DqElement', function() {
     });
   });
 
   describe('selector', function() {
     it('should prefer selector from spec object', function() {
-      fixture.innerHTML = '<div id="foo" class="bar">Hello!</div>';
-      var result = new DqElement(
-        fixture.firstChild,
-        {},
-        {
-          selector: 'woot'
-        }
-      );
+      var vNode = queryFixture('<div id="target" class="bar">Hello!</div>');
+      var spec = { selector: 'woot' };
+      var result = new DqElement(vNode, {}, spec);
       assert.equal(result.selector, 'woot');
     });
   });
 
   describe('ancestry', function() {
     it('should prefer selector from spec object', function() {
-      fixture.innerHTML = '<div id="foo" class="bar">Hello!</div>';
-      var result = new DqElement(
-        fixture.firstChild,
-        {},
-        {
-          ancestry: 'woot'
-        }
-      );
+      var vNode = queryFixture('<div id="target" class="bar">Hello!</div>');
+      var spec = { ancestry: 'woot' };
+      var result = new DqElement(vNode, {}, spec);
       assert.equal(result.ancestry, 'woot');
     });
   });
 
   describe('xpath', function() {
     it('should prefer selector from spec object', function() {
-      fixture.innerHTML = '<div id="foo" class="bar">Hello!</div>';
-      var result = new DqElement(
-        fixture.firstChild,
-        {},
-        {
-          xpath: 'woot'
-        }
-      );
+      var vNode = queryFixture('<div id="target" class="bar">Hello!</div>');
+      var spec = { xpath: 'woot' };
+      var result = new DqElement(vNode, {}, spec);
       assert.equal(result.xpath, 'woot');
     });
   });
 
   describe('absolutePaths', function() {
     it('creates a path all the way to root', function() {
-      fixtureSetup('<div id="foo" class="bar">Hello!</div>');
-
-      var result = new DqElement(fixture.firstChild, {
+      var vNode = queryFixture('<div id="target" class="bar">Hello!</div>');
+      var result = new DqElement(vNode, {
         absolutePaths: true
       });
       assert.include(result.selector[0], 'html > ');
       assert.include(result.selector[0], '#fixture > ');
-      assert.include(result.selector[0], '#foo');
+      assert.include(result.selector[0], '#target');
+    });
+  });
+
+  describe('nodeIndexes', function() {
+    it('is taken from virtualNode', function() {
+      fixtureSetup('<i></i><b></b><s></s>');
+      assert.deepEqual(new DqElement(fixture.children[0]).nodeIndexes, [1]);
+      assert.deepEqual(new DqElement(fixture.children[1]).nodeIndexes, [2]);
+      assert.deepEqual(new DqElement(fixture.children[2]).nodeIndexes, [3]);
+    });
+
+    it('is taken from spec, over virtualNode', function() {
+      var vNode = queryFixture('<div id="target"></div>');
+      var spec = { nodeIndexes: [123, 456] };
+      var dqElm = new DqElement(vNode, {}, spec);
+      assert.deepEqual(dqElm.nodeIndexes, [123, 456]);
+    });
+
+    it('is [] when the element is unknown.', function() {
```

Comment:
I didn't know what an unknown element was so had to look back into the code to figure it out.

```suggestion
    it('is [] when the element is not in the virtual tree.', function() {
```